builder/triton/ssh.go

```diff
@@ -0,0 +1,36 @@
+package triton
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/multistep"
+	"golang.org/x/crypto/ssh"
+)
+
+func commHost(state multistep.StateBag) (string, error) {
+	driver := state.Get("driver").(Driver)
+	machineID := state.Get("machine").(string)
+
+	machine, err := driver.GetMachine(machineID)
+	if err != nil {
+		return "", err
+	}
+
+	return machine, nil
+}
+
+func sshConfig(state multistep.StateBag) (*ssh.ClientConfig, error) {
+	config := state.Get("config").(Config)
+
+	signer, err := ssh.ParsePrivateKey([]byte(config.Comm.SSHPrivateKey))
+	if err != nil {
+		return nil, fmt.Errorf("Error setting up SSH config: %s", err)
+	}
+
+	return &ssh.ClientConfig{
+		User: config.Comm.SSHUsername,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+	}, nil
```

Comment:
@jen20 Good point. So basically the following types (or brands as they are called in Triton / Joyent lingo) are available:

* Joyent / Smartmachine (Basically an Illumos / Solaris zone)
  * Authentication for VM's is performed with the special SSH plugin as described above.
* KVM (Full blown virtualization)
  * I've tested this with the `ubuntu-certified-15.04` image. The users public key which is in Triton is automatically added to the `authorized-keys` file for the `root` user (or the `ubuntu` user in case of the Ubuntu image). I also did a test with the `freebsd-11` image which works the same.  The tools used to provision the SSH keys in *nix KVM images can for example be found in the source repo of the FreeBSD image: https://github.com/joyent/freebsd-kvm-image-builder/tree/master/triton-freebsd-guesttools/lib/smartdc. The Windows KVM images are only available on the Joyent public cloud so I couldn't test those. 
* LX (Docker API + Linux ABI compatibility layer)
  *  As far as I can tell there is no point in supporting this since it should work with the Packer docker builder (altough I haven't tested it).

So with the current SSH setup everything except for KVM Windows images will just work as far as I can tell.